server/index.js

```diff
@@ -33,8 +33,30 @@ app.get('/api/products', (req, res, next) => {
     .catch(err => next(err));
 });
 
+app.get('/api/products/:productId', express.json(), (req, res, next) => {
+  const id = parseInt(req.params.productId, 10);
+  const selectId = `
+    select "name",
+           "price",
+           "image",
+           "shortDescription",
+           "longDescription",
+           "productId"
+      from "products"
+     where "productId" = $1;
+  `;
+  db.query(selectId, [id])
+    .then(result => {
+      if (!result.rows[0]) {
+        next(new ClientError('The product id does not exist', 400));
+      }
+      res.json(result.rows[0]);
```

Comment:
Be sure to include either an `else` or a `return` to ensure that the server can only respond one of two ways.